Various fixes, mostly systemd-related

0xC0ncord opened the pull request:
Various scattered fixes, mostly systemd-related.

Some of these are:
- Allow init to `nnp_transition` to any init daemon domain (systemd's `NoNewPrivileges` unit option)
- Allowing domains which can use nsswitch to read resolvconf symlinks (for systemd-resolved's stub configuration)
- Polyinstantiation fixes
- Various dontaudits

pebenito commented on policy/modules/services/networkmanager.te:
```diff
@@ -192,6 +192,10 @@ userdom_write_user_tmp_sockets(NetworkManager_t)
 userdom_dontaudit_use_user_ttys(NetworkManager_t)
 
+ifdef(`init_systemd',`
+	init_get_system_status(NetworkManager_t)
```
I don't think the ifdef is needed.

pebenito commented on policy/modules/system/init.fc:
```diff
@@ -7,6 +7,8 @@
 /etc/rc\.d/init\.d/.*	--	gen_context(system_u:object_r:initrc_exec_t,s0)
 /etc/sysconfig/network-scripts/ifup-ipsec -- gen_context(system_u:object_r:initrc_exec_t,s0)
 
+/etc/systemd/system(/.*)?	gen_context(system_u:object_r:systemd_unit_t,s0)
```
This should already be handled by the `file_contexts.subs_dist`.

0xC0ncord replied:
You are right. I think this is just a simple oversight.

pebenito commented on policy/modules/system/sysnetwork.fc:
```diff
@@ -21,6 +21,7 @@ ifdef(`distro_debian',`
 /etc/hostname		--	gen_context(system_u:object_r:net_conf_t,s0)
 /etc/denyhosts.*	--	gen_context(system_u:object_r:net_conf_t,s0)
 /etc/resolv\.conf.*	--	gen_context(system_u:object_r:net_conf_t,s0)
+/etc/resolv\.conf.*	-l	gen_context(system_u:object_r:net_conf_t,s0)
```
Normally we leave symlinks as the general label, `etc_t` in this case.

0xC0ncord commented:
Sorry for the push after the approval.

I fixed a small conflict with rules added to `systemd_sessions_t` since a rule was added allowing it to `getattr` on all filesystems, so I just removed the rules I added for specific ones.
Also fixed the merge conflict on `sudo.if`.

pebenito commented on policy/modules/system/systemd.te:
```diff
@@ -671,6 +674,7 @@ sysnet_manage_config(systemd_hostnamed_t)
 systemd_log_parse_environment(systemd_hostnamed_t)
 
 # Allow reading /run/udev/data/+dmi:id
+udev_search_runtime(systemd_hostnamed_t)
 udev_read_runtime_files(systemd_hostnamed_t)
```
I guess I missed this before, but this indicates that `udev_read_runtime_files()` is missing perms, since it should already provide `udev_search_runtime()`.

0xC0ncord replied:
That appears to be the case. I will move this access into that interface.

pebenito commented on policy/modules/system/systemd.te:
```diff
@@ -1451,9 +1466,8 @@ allow systemd_sessions_t systemd_sessions_runtime_t:file manage_file_perms;
+fs_getattr_cgroup(systemd_sessions_t)
 fs_search_cgroup_dirs(systemd_sessions_t)
-fs_search_tmpfs(systemd_sessions_t)
-fs_search_ramfs(systemd_sessions_t)
```
I don't understand the deletions. There's the getattr all, but no search all. I think you meant to remove `fs_getattr_cgroup(systemd_sessions_t)` from your patch?

0xC0ncord replied:
Ah yes, sorry for the confusion. That is exactly what I wanted to do.

pebenito commented on policy/modules/system/udev.if:
```diff
@@ -503,6 +503,7 @@ interface(`udev_read_runtime_files',`
 	')
 
 	files_search_runtime($1)
+	udev_search_runtime($1)
 	read_files_pattern($1, udev_runtime_t, udev_runtime_t)
```
Hmm, because of the `read_files_pattern()` there should already be the equivalent of `udev_search_runtime()` access in this interface.

0xC0ncord replied:
Hmm. Unfortunately I didn't save the AVC for why I added this. I think for now I'll drop it and if I can reproduce it in the future I will look into it further.

pebenito commented on policy/modules/system/systemd.te:
```diff
@@ -1377,6 +1387,10 @@ init_runtime_filetrans(systemd_resolved_t, systemd_resolved_runtime_t, dir)
 
 dev_read_sysfs(systemd_resolved_t)
 
+fs_getattr_cgroup(systemd_resolved_t)
```
Please double check the fs getattr rules, as some of the domains have `fs_getattr_all()` but unfortunately the lines are out of order.

0xC0ncord replied:
Good call. There were some duplicate rules added since the `fs_getattr_all()`s were added some time after I originally worked on this.

0xC0ncord commented:
I think I pushed changes while Github was having internal issues which could explain why the checks aren't running.
